src/ChannelServer/Network/Handlers/Parties.cs

```diff
@@ -0,0 +1,308 @@
+
+            if (creature == null)
+            {
+                Send.PartyLeaveR(creature, false);
+                return;
+            }
+
+            if (!creature.IsInParty)
+            {
+                Send.PartyLeaveR(creature, false);
+                return;
+            }
+
+            var party = creature.Party;
+
+            // TODO: Check if there are other regions or situations in which you cannot leave a party.
+            var canLeave = !creature.Region.IsDynamic;
+
+            if (canLeave)
+            {
+                creature.Party.RemoveMember(creature);
+
+
+                if (party.TotalMembers > 0)
+                {
+                    Send.PartyLeaveUpdate(creature, party);
+
+                    if (party.IsOpen)
+                        Send.PartyMemberWantedRefresh(party);
+
+                    // What is this?
+                    //Send.PartyWindowUpdate(creature, party);
+
+                    if (party.Leader == creature)
+                    {
+                        party.SetLeader(party.GetNextLeader());
+                        if (party.IsOpen)
+                            party.Close();
+                        Send.PartyChangeLeader(party);
+                    }
+                }
+                else
+                {
+                    if (party.IsOpen)
+                        party.Close();
+                }
+            }
+
+            Send.PartyLeaveR(creature, canLeave);
+        }
+
+        [PacketHandler(Op.PartyRemove)]
+        public void RemoveFromParty(ChannelClient client, Packet packet)
+        {
+            var creature = client.GetCreatureSafe(packet.Id);
+
+            if (creature == null)
+                return;
+
+            if (!creature.IsInParty)
+                return;
+
+            var party = creature.Party;
+            if (party.Leader != creature)
+                return;
+
+            
+            var canRemove = !creature.Region.IsDynamic;
+            if (canRemove)
+            {
+                var target = party.ContainsMember(packet.GetLong());
+                if (target != null && (target != creature))
+                {
+                    party.RemoveMember(target);
+                    Send.PartyRemoved(target);
+                    Send.PartyLeaveUpdate(target, party);
+                }
+                else
+                    canRemove = false;
+            }
+
+            Send.PartyRemoveR(creature, canRemove);
+        }
+
+        [PacketHandler(Op.PartyChangeSetting)]
+        public void PartyChangeSettings(ChannelClient client, Packet packet)
+        {
+            var creature = client.GetCreatureSafe(packet.Id);
+            if (creature == null)
+                return;
+
+            if (!creature.IsInParty)
+                return;
+
+            var party = creature.Party;
+            if (creature != party.Leader)
+                return;
+
+            party.ChangeSettings(packet);
```

Comment:
EW... packet parsing should be done in the handlers, never anywhere else, this is not legacy!